src/StringUtils.cpp

```diff
@@ -14,6 +14,223 @@
+
+template<>
+std::string NAS2D::stringTo(const std::string& value)
+{
+	return value;
+}
+
+template<>
+bool NAS2D::stringTo(const std::string& value)
+{
+	auto v = toLowercase(value);
+	if (v == "true") return true;
+	if (v == "false") return false;
+	throw std::invalid_argument("Value is not 'true' or 'false' value: " + value);
+}
+
+template<>
+char NAS2D::stringTo(const std::string& value)
+{
+	return detail::rangeCheckHelper<char>(value);
+}
+
+template<>
+unsigned char NAS2D::stringTo(const std::string& value)
+{
+	return detail::rangeCheckHelper<unsigned char>(value);
+}
+
+template<>
+signed char NAS2D::stringTo(const std::string& value)
+{
+	return detail::rangeCheckHelper<signed char>(value);
+}
+
+template<>
+short NAS2D::stringTo(const std::string& value)
+{
+	return detail::rangeCheckHelper<short>(value);
+}
+
+template<>
+unsigned short NAS2D::stringTo(const std::string& value)
+{
+	return detail::rangeCheckHelper<unsigned short>(value);
+}
+
+template<>
+int NAS2D::stringTo(const std::string& value)
+{
+	return std::stoi(value);
+}
+
+template<>
+unsigned int NAS2D::stringTo(const std::string& value)
+{
+	const auto valueAsInteger = std::stoul(value);
+	if constexpr (std::numeric_limits<unsigned int>::max() < std::numeric_limits<unsigned long>::max())
+	{
+		if (std::numeric_limits<unsigned int>::max() < valueAsInteger)
+		{
+			throw std::out_of_range("from_string argument out of range");
+		}
+	}
+	return static_cast<unsigned int>(valueAsInteger);
+}
+
+template<>
+long NAS2D::stringTo(const std::string& value)
+{
+	return std::stol(value);
+}
+
+template<>
+unsigned long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoul(value);
+}
+
+template<>
+long long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoll(value);
+}
+
+template<>
+unsigned long long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoull(value);
+}
+
+template<>
+float NAS2D::stringTo(const std::string& value)
+{
+	return std::stof(value);
+}
+
+template<>
+double NAS2D::stringTo(const std::string& value)
+{
+	return std::stod(value);
+}
+
+template<>
+long double NAS2D::stringTo(const std::string& value)
+{
+	return std::stold(value);
+}
+
+
+template<>
+std::string NAS2D::stringFrom(std::string value)
+{
+	return value;
+}
+
+template<>
+std::string NAS2D::stringFrom(bool value)
+{
+	return value ? "true" : "false";
+}
+
+template<>
+std::string NAS2D::stringFrom(char value)
+{
+	return std::string{value};
+}
+
+template<>
+std::string NAS2D::stringFrom(signed char value)
+{
+	return std::to_string(static_cast<int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned char value)
+{
+	return std::to_string(static_cast<unsigned int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(short value)
+{
+	return std::to_string(static_cast<int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned short value)
+{
+	return std::to_string(static_cast<unsigned int>(value));
+}
```

Comment:
I'm not sure the cast adds much value here. Automatic [numeric promotion](https://en.cppreference.com/w/cpp/language/implicit_conversion) should allow an appropriate function overload to be called.

Removing the cast makes the implementation match the others, and can allow further consolidation of the function definitions.

Reply:
`std::to_string` doesn't have overloads for types smaller than `int`, hence, the casts.

Reply:
And I'm saying it doesn't need a cast. If you try to pass a `short`, it will be automatically promoted to `int`, and then call the `int` overload.